### PR TITLE
Aligned requirements with upstream 2.6.1

### PR DESCRIPTION
**Pull request by timostrunk**
Upstream pyproject.toml changed a lot:
https://github.com/rthalley/dnspython/blob/main/pyproject.toml
As far as I understand nothing is required anymore. I kept the sniffio dependency, because I don't want to break current users and sniffio is used in the asyncio check.

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by timostrunk** [comment]
@conda-forge-admin, please rerender

**Comment by synapticarbors** [comment]
@timostrunk -- good catch. I guess the grayskull update doesn't catch `run_constrained` requirements? I think this looks good, but will wait for the other co-maintainers to chime in before merging.

**Comment by xylar** [comment]
I think we should remote bot automerge to prevent this kind of thing happening.

**Comment by xylar** [comment]
https://github.com/timostrunk/dnspython-feedstock/blob/fix_requirements/conda-forge.yml#L3

**Comment by xylar** [comment]
My guess is that the grayskull bot update failed because grayskull is still struggling with certain `pyproject.toml` syntax (lists of dependencies with different constraints https://github.com/conda/grayskull/issues/463).

**Comment by synapticarbors** [comment]
@timostrunk -- would you remove the automerge from the config shown in https://github.com/conda-forge/dnspython-feedstock/pull/18#issuecomment-1969687678? Then I think we should be good to merge this PR.

**Comment by timostrunk** [comment]
Will do

**Comment by timostrunk** [comment]
@conda-forge-admin, please rerender

File: conda-forge.yml
Comment (timostrunk) [inline conda-forge.yml:3]: Automerge is removed and I rerendered afterwards

**Review by xylar — APPROVED** [review]

**Comment by xylar** [comment]
Thanks @timostrunk!

**Pull request merged**
